lib/mix/lib/mix/dep/lock.ex

```diff
@@ -62,4 +63,13 @@ defmodule Mix.Dep.Lock do
   defp lockfile do
     Mix.Project.config[:lockfile]
   end
+
+  defp assert_no_merge_conflicts_in_lockfile(lockfile, info) do
+    if Regex.match?(~r/^(?:<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|)/, info) do
+      Mix.raise """
+      Your #{lockfile} contains merge conflicts.
+               Run `git checkout HEAD -- #{lockfile}` first to get a clean lock.
```

Comment:
I don't think that's necessary; most of Elixir's source either uses `"""` and doesn't care about output like

```
** (Mix) Your mix.lock contains merge conflicts.
Run `git checkout HEAD -- mix.lock` first to get a clean lock.
```

(which is what I prefer personally), or uses no newlines at all and lets the host terminal do the splitting :).